apps/glusterfs/brick_allocate.go

```diff
@@ -24,10 +24,23 @@ type BrickSet struct {
 	Bricks  []*BrickEntry
 }
 
+// NewBrickSet creates an empty brick set tracking object with
+// the expected final size of s.
 func NewBrickSet(s int) *BrickSet {
 	return &BrickSet{SetSize: s, Bricks: []*BrickEntry{}}
 }
 
+// NewSparseBrickSet creates a brick set tracking object with
+// the expected final size of s and the brick tracking slice
+// pre-sized to s. This means that the array can be directly
+// indexed but may contains null brick entries.
+func NewSparseBrickSet(s int) *BrickSet {
+	return &BrickSet{
+		SetSize: s,
+		Bricks:  make([]*BrickEntry, s),
```

Comment:
I'm new to golang, but wouldn't it be better style, to make a new type from []*BrickEntry, like `type BrickSlice = []*BrickEntry` and use that instead?

Reply:
It is certainly do-able... I just don't see the value of adding that now ATM. If we were to do it I'd add it to the file where BrickEntry was defined.